pkgs/by-name/he/hello/package.nix

```diff
@@ -1,46 +1,41 @@
-{ callPackage
-, lib
-, stdenv
-, fetchurl
-, nixos
-, testers
-, hello
-}:
+{ mkPackageWithDeps, layers }: mkPackageWithDeps ({ stdenv, fetchurl, testers, lib, callPackage, nixos }: [
```

Comment:
Consider adding `layers` as well as a function `overrideLayers` to the call here, such that they don't come from the normal fixpoint. Otherwise it's not possible to override that `layers`, which someone will inevitably need. 

It also makes the separation of concerns a bit nicer:
```nix
# 1 parameter to declare that we'll be doing it the layered way
{ mkPackageWithDeps }: mkPackageWithDeps ({
  # all the rest goes here
  layers, ...
}: [
  # ...
])
```

That changes the fixpoint "state" from just overlays to `{ overlays = <composition of functions>; layers = <attrset of functions brought into scope by mkPackageWithDeps>; }`, but that seems manageable and not something users have to be aware of, as long as it works correctly. It's still just a single fixpoint, so it seems feasible to get it right.